File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/gui/components/BaseLayout.java
package pize.tests.minecraft.game.gui.components;

import pize.graphics.util.batch.TextureBatch;
import pize.gui.UIComponent;
import pize.gui.constraint.Constraint;

public class BaseLayout extends UIComponent<TextureBatch>{
    
    public BaseLayout(){
        super.setSize(Constraint.matchParent(), Constraint.matchParent());
    }
    
    @Override
    protected void render(TextureBatch batch, float x, float y, float width, float height){
        // if(isHover()){
        //     batch.drawQuad(0.1, x, y, width, height);
        //     batch.resetColor();
        // }
    }
    
}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/gui/components/BaseLayout.kt
package pize.tests.minecraft.game.gui.components

import pize.graphics.util.batch.TextureBatch
import pize.gui.UIComponent
import pize.gui.constraint.Constraint.Companion.matchParent

class BaseLayout : UIComponent<TextureBatch?>() {
    init {
        super.setSize(matchParent(), matchParent())
    }

    protected override fun render(batch: TextureBatch, x: Float, y: Float, width: Float, height: Float) {
        // if(isHover()){
        //     batch.drawQuad(0.1, x, y, width, height);
        //     batch.resetColor();
        // }
    }
}